### ISSUE TITLE
[3.0] Label text translation is not working

### ISSUE THREAD

**Issue by hickop** [godotengine/godot#8765]
**Operating system or device - Godot version:**
3.0 master on Archlinux

**Issue description:**
I'm having a crash when translating a label from a long text to a shorter one with error:
```
ERROR: operator[]: Condition ' p_index < 0 || p_index >= size() ' is true. returned: aux
   At: core/vector.h:139.
```

The text is cut when translating a label from a short text to a longer one. Has same quantity of chars as previous text.

**Steps to reproduce:**
Using TranslationServer.set_locale() to change localization on label.

**Link to minimal example project:**
[godot-translation-test.zip](https://github.com/godotengine/godot/files/999605/godot-translation-test.zip)

**Comment by bojidar-bg** [comment]
Some GDB backtraces (from a week-old godot3 version):
```cpp
Thread 1 "godot3.dbg" hit Breakpoint 1, Vector<wchar_t>::operator[] (this=0x3eee940, p_index=6) at core/vector.h:139
139                             ERR_FAIL_COND_V(p_index < 0 || p_index >= size(), aux);
(gdb) bt
#0  Vector<wchar_t>::operator[] (this=0x3eee940, p_index=6) at core/vector.h:139
#1  0x000000000183070b in Label::_notification (this=0x3eee580, p_what=30) at scene/gui/label.cpp:258
#2  0x00000000016d130c in Label::_notificationv (this=0x3eee580, p_notification=30, p_reversed=false) at ./scene/gui/label.h:39
#3  0x0000000002030e9c in Object::notification (this=0x3eee580, p_notification=30, p_reversed=false) at core/object.cpp:941
#4  0x0000000001a9b31f in CanvasItem::_update_callback (this=0x3eee580) at scene/2d/canvas_item.cpp:145
#5  0x0000000000c91738 in MethodBind0::call (this=0x3b5a240, p_object=0x3eee580, p_args=0x0, p_arg_count=0, r_error=...) at core/method_bind.inc:47
#6  0x0000000002030d18 in Object::call (this=0x3eee580, p_method=..., p_args=0x0, p_argcount=0, r_error=...) at core/object.cpp:931
#7  0x0000000001fe5d16 in MessageQueue::_call_function (this=0x321af20, p_target=0x3eee580, p_func=..., p_args=0x7ffff7e12068, p_argcount=0, p_show_error=false)
    at core/message_queue.cpp:300
#8  0x0000000001fe5f31 in MessageQueue::flush (this=0x321af20) at core/message_queue.cpp:338
#9  0x00000000016f5f95 in SceneTree::input_event (this=0x3e49e30, p_event=...) at scene/main/scene_main_loop.cpp:480
#10 0x0000000000c399f8 in InputDefault::parse_input_event (this=0x3966f70, p_event=...) at main/input_default.cpp:394
#11 0x0000000000c2ed28 in OS_X11::process_xevents (this=0x7fffffffe010) at platform/x11/os_x11.cpp:1368
#12 0x0000000000c30ee9 in OS_X11::run (this=0x7fffffffe010) at platform/x11/os_x11.cpp:1913
#13 0x0000000000c295b0 in main (argc=1, argv=0x7fffffffe528) at platform/x11/godot_x11.cpp:49
(gdb) cont
Continuing.
ERROR: operator[]: Condition ' p_index < 0 || p_index >= size() ' is true. returned: aux
   At: core/vector.h:139.

Thread 1 "godot3.dbg" received signal SIGSEGV, Segmentation fault.
0x000000000183070b in Label::_notification (this=0x3eee580, p_what=30) at scene/gui/label.cpp:258
258                                                     CharType n = xl_text[i + pos + 1];
(gdb) bt
#0  0x000000000183070b in Label::_notification (this=0x3eee580, p_what=30) at scene/gui/label.cpp:258
#1  0x00000000016d130c in Label::_notificationv (this=0x3eee580, p_notification=30, p_reversed=false) at ./scene/gui/label.h:39
#2  0x0000000002030e9c in Object::notification (this=0x3eee580, p_notification=30, p_reversed=false) at core/object.cpp:941
#3  0x0000000001a9b31f in CanvasItem::_update_callback (this=0x3eee580) at scene/2d/canvas_item.cpp:145
#4  0x0000000000c91738 in MethodBind0::call (this=0x3b5a240, p_object=0x3eee580, p_args=0x0, p_arg_count=0, r_error=...) at core/method_bind.inc:47
#5  0x0000000002030d18 in Object::call (this=0x3eee580, p_method=..., p_args=0x0, p_argcount=0, r_error=...) at core/object.cpp:931
#6  0x0000000001fe5d16 in MessageQueue::_call_function (this=0x321af20, p_target=0x3eee580, p_func=..., p_args=0x7ffff7e12068, p_argcount=0, p_show_error=false)
    at core/message_queue.cpp:300
#7  0x0000000001fe5f31 in MessageQueue::flush (this=0x321af20) at core/message_queue.cpp:338
#8  0x00000000016f5f95 in SceneTree::input_event (this=0x3e49e30, p_event=...) at scene/main/scene_main_loop.cpp:480
#9  0x0000000000c399f8 in InputDefault::parse_input_event (this=0x3966f70, p_event=...) at main/input_default.cpp:394
#10 0x0000000000c2ed28 in OS_X11::process_xevents (this=0x7fffffffe010) at platform/x11/os_x11.cpp:1368
#11 0x0000000000c30ee9 in OS_X11::run (this=0x7fffffffe010) at platform/x11/os_x11.cpp:1913
#12 0x0000000000c295b0 in main (argc=1, argv=0x7fffffffe528) at platform/x11/godot_x11.cpp:49
```

**Comment by hickop** [comment]
fixed with db3b05d2893dcaddeb3bcb10b845ff150eb50895
